packages/eas-cli/src/project/customBuildConfig.ts

```diff
@@ -24,6 +31,13 @@ export async function validateCustomBuildConfigAsync(
       `Custom build configuration file ${chalk.bold(relativeConfigPath)} does not exist.`
     );
   }
+  if (await vcsClient.isFileIgnoredAsync(relativeConfigPath)) {
```

Comment:
May a config file reference other files? Maybe not now, but in the future? Or TypeScript functions? Should we make sure whole `.eas` is committed instead (instead of just the config file)?

Reply:
>May a config file reference other files?

https://docs.expo.dev/custom-builds/schema/#import

>Or TypeScript functions?

https://docs.expo.dev/custom-builds/schema/#functionsfunction_namepath

>Should we make sure whole .eas is committed instead (instead of just the config file)?

Good question 🤔 

I believe that we should fail if the currently used config file is git ignored, but maybe display a warning if other files in `.eas` are git ignored? Ideally, I feel like we can try to resolve which custom functions will be needed for a given build and which config files will be needed and validate that they are not git ignored, but I think it will require some more changes, and I think that we can merge it as v1 to make it better and not wait with it for other stuff to come along. WDYT?